db/redis.go

```diff
@@ -256,20 +255,31 @@ func (r *RedisDriver) GetSimilarCpesByTitle(query string, n int, algorithm edlib
 	}
 
 	ctx := context.Background()
+	t, err := r.conn.Type(ctx, titleListKey).Result()
+	if err != nil {
+		return nil, xerrors.Errorf("Failed to TYPE CPE#Titles. err: %w", err)
+	}
+
 	var ts []string
-	bs, err := r.conn.Get(ctx, titleListCacheKey).Bytes()
-	if err == nil {
+	switch t {
+	case "string":
+		bs, err := r.conn.Get(ctx, titleListKey).Bytes()
+		if err != nil {
+			return nil, xerrors.Errorf("Failed to Get Titles. err: %w", err)
+		}
 		if err := json.Unmarshal(bs, &ts); err != nil {
 			return nil, xerrors.Errorf("Failed to Unmarshal JSON. err: %w", err)
 		}
-	} else {
-		if !xerrors.Is(err, redis.Nil) {
-			return nil, xerrors.Errorf("Failed to Get Titles. err: %w", err)
-		}
-		ts, err = r.conn.SMembers(ctx, titleListKey).Result()
+	case "set":
```

Comment:
Only reading the latest source code, one can't know why this branch exists.
How about adding comment (e.g. for backward compatibility) for future source code readers (us, 99%) ?

Reply:
https://github.com/vulsio/go-cpe-dictionary/compare/82b51ab3fa2074c38df329f589288ccffe603835..a1b7351fdd881074f7004df4e84111619e28edbc#diff-47497c0e5aee216ce040118072f4e2199cc00ec8f2c55059b2c4729ec026e101R273